### ISSUE TITLE
Many false positives in a custom SSD model with Tensorflow object detection API

### ISSUE THREAD

**Issue by YonitZall** [tensorflow/tensorflow#29078]
My model has 2 classes (no background class) and is trained using transfer learning with ssd_mobilenet_v2_coco. It detects and classifies well the objects it was trained on. However, on new images it also detects many false positive bounding box's of the background. Sometimes the background objects that are detected are objects that existed in the coco data set (e.g. a cup), but not only - also a black screen is detected as an object.

Can it be that my model features are still sensitive to the Coco data set objects?
I think that on the config file, the relevant part for this issue is the hard_example_miner. However I did not understand how is this related and if I should change my current configuration.
This is the configuration I'm using:

hard_example_miner {

num_hard_examples: 3000

iou_threshold: 0.99

loss_type: CLASSIFICATION

max_negatives_per_positive: 3

in_negatives_per_image: 3

}

Should I make any changes to this? Or any other solutions to improve the model on background images?

**Comment by muddham** [comment]
@YonitZall This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose). Thanks!

**Comment by muddham** [comment]
@YonitZall Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by selmankaplan** [comment]
> My model has 2 classes (no background class) and is trained using transfer learning with ssd_mobilenet_v2_coco. It detects and classifies well the objects it was trained on. However, on new images it also detects many false positive bounding box's of the background. Sometimes the background objects that are detected are objects that existed in the coco data set (e.g. a cup), but not only - also a black screen is detected as an object.
> 
> Can it be that my model features are still sensitive to the Coco data set objects?
> I think that on the config file, the relevant part for this issue is the hard_example_miner. However I did not understand how is this related and if I should change my current configuration.
> This is the configuration I'm using:
> 
> hard_example_miner {
> 
> num_hard_examples: 3000
> 
> iou_threshold: 0.99
> 
> loss_type: CLASSIFICATION
> 
> max_negatives_per_positive: 3
> 
> in_negatives_per_image: 3
> 
> }
> 
> Should I make any changes to this? Or any other solutions to improve the model on background images?

hi,
did you find any method other than create more class for solve this problem?
Thanks

**Comment by apramanik62** [comment]
@YonitZall 
did you get any solutions? I am facing the same problem

**Comment by selmankaplan** [comment]
There are different parameters that affect this, but when I did enough training with appropriate learning rate values, I was able to train a model with higher fps and accuracy than the faster rcnn model. You have to find out for yourself which parameter should be edited. Although it is not certain, as a general solution, it is also used as a solution to train in more than one class and only show the object you are looking for on the screen.

**Comment by apramanik62** [comment]
> There are different parameters that affect this, but when I did enough training with appropriate learning rate values, I was able to train a model with higher fps and accuracy than the faster rcnn model. You have to find out for yourself which parameter should be edited. Although it is not certain, as a general solution, it is also used as a solution to train in more than one class and only show the object you are looking for on the screen.

Hello @selmankaplan 
Thanks for the reply.
Can you tell me which parameters should I look on ? And on which file will I get those parameters.
I have used ssd pet config and I am working on bird detection. I have used pretrained model of  the Coco data  and find model is detecting bird as bird in some cases. Any idea?

**Comment by selmankaplan** [comment]
If you're training using tensorflow, your train and test images and its label frames are effective parameters for your training data. If you created the dataset properly, the training parameters are included in the .config file. First of all, you should pay attention to the learning rate parameter and data augmentation options parameters. Although data augmentation parameters seemed to be a good choice, there were negative effects while training. I would like to state that I do not know whether the labels size is data augmentation or not when pictures data augmentation. Finally, you should make sure that you finish the training at the appropriate time. In this way, you can train in a simple train.
If you did a single-class training and get a person output instead of a bird, you can set this in your .bptxt file instead of the training. If you have done a training with more than one class, I recommend that you review your education from the beginning.
